django/db/backends/oracle/validation.py

```diff
@@ -19,4 +22,30 @@ def check_field_type(self, field, field_type):
                     id="fields.W162",
                 )
             )
+
+        if (
+            field_type.startswith("number")
+            and hasattr(field, "max_digits")
+            and hasattr(field, "decimal_places")
+        ):
+            if field.max_digits > 38:
+                errors.append(
+                    checks.Warning(
+                        "%s does not support DecimalField with max_digits > 38."
+                        % self.connection.display_name,
+                        obj=field,
+                        id="oracle.W001",
+                    )
+                )
+
+            if field.decimal_places > 127:
```

Comment:
Any reason not to create, per database backend, constants variables at module or class scopes for the values used to verify the `max_digits` and `decimal_places`, and then reuse them in validation and messages?

Okay, maybe the only benefit of this is the reduction in duplication of hardcoded numbers in the codebase and messages and the reduction in the number of places that need to be changed if a new database version changes it limits.

Another question is whether the constant variables should be in the public API or in the private API of the database backend.